src/middleware/otpauthMiddleware.ts

```diff
@@ -0,0 +1,64 @@
+import randomstring from 'randomstring';
+import Jwt, { VerifyErrors } from 'jsonwebtoken';
+import { Request, Response, NextFunction } from 'express';
+import { config } from 'dotenv';
+import { sendEmail } from '../utils/email';
+import { DataInfo } from '../controllers/OTPcontroller';
+import { decodeToken, verifyToken } from '../utils/tokenGenerate';
+
+
+config();
+
+
+export const generateOtpToken = async (id: string, email:string) => {
+  const code = await randomstring.generate({
+    length: 6,
+    charset: 'numeric',
+  });
+  const mailOptions = {
+    to: email,
+    subject: 'OTP Email verification',
+    template: 'otp',
+    context: {
+      otp: `${code}`,
+    },
+  };
+  await sendEmail(mailOptions);
+  const body = {
+    id,
+    email,
+    otp: code,
+  };
+  const secret = process.env.APP_SECRET;
+  const token = await Jwt.sign({ body }, secret as string, {
+    expiresIn: '180s',
+  });
```

Comment:
we have this already implemented assigning token

Reply:
but the payload is different  here i nclude otp too but for the function we have for assigning token have only two arguments!!!